### DISCUSSION TITLE
work around for AUROC metrics error `ValueError: No positive samples in targets, true positive value should be meaningless` during sanity check or dev runs?

### DISCUSSION THREAD

**Discussion by jmerkow** [Lightning-AI/torchmetrics#581] [Classification] [answered]
I have a multi-label problem with a rare class, I want to use AUROC metric to generate per-class metrics.  I get `ValueError: No positive samples in targets, true positive value should be meaningless` error during sanity checks and fast dev runs, for obvious reasons.  Is there a way to get these metric to return something here (even know I know it meaningless)?

**Comment by SkafteNicki** [comment, marked as answer]
Hi @jmerkow,
Thanks for asking this question. Because of your feedback and many others, we decided on changing this behaviour in this PR https://github.com/PyTorchLightning/metrics/pull/583. Instead you should only get an warning, and in the case of AUROC it will return a score when either the positive or negative class is completely missing.
